### PR TITLE
Improve typos configuration

### PR DESCRIPTION
**Pull request by szepeviktor**
## Description

Make typos config more strict preventing future ignore of actual typos. Put everything in its place.

You could check hidden files by removing `.` from the command line.
https://github.com/static-web-server/static-web-server/blob/9afff97acd15320bbd8b7c426d42a3fd369992c3/Makefile#L301

### CODE REVIEW

File: .github/workflows/config/typos.toml [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
-ba = "ba"
-caf = "caf"
-flate = "flate"
-Ody = "Ody"
-prefered = "prefered"
+compressable = "compressable"
+
+[default.extend-identifiers]
+flate2 = "flate2"
+OdyX = "OdyX"
```
Comment (joseluisq) [inline .github/workflows/config/typos.toml]:
```suggestion
```

File: .github/workflows/config/typos.toml [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+extend-exclude = [
+    ".git/",
+]
+ignore-hidden = false
+
+[default]
+extend-ignore-re = [
+    "\\[[0-9a-f]{7}\\]",
+    "commit/[0-9a-f]{7}",
+    "\\{a,ba,da,k,z\\}sh",
```
Comment (joseluisq) [inline .github/workflows/config/typos.toml:11]: Let's add this to catch up with GitHub usernames and remove line 19.

```suggestion
    "\\{a,ba,da,k,z\\}sh",
    "\\[@[a-zA-Z0-9\\-]+\\]",
```

**Review by joseluisq — CHANGES_REQUESTED** [review]
Looks fine, just a minor change to do.

**Comment by szepeviktor** [comment]
Done 🍏

**Review by joseluisq — APPROVED** [review]
Thanks!

**Comment by szepeviktor** [comment]
Glad to contribute.

**Pull request merged**